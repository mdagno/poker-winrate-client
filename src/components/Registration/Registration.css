#registration-form {
  display: flex;
  flex-direction: column;
  color: #222;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
  margin-top: 11%;
  background-color: #ccc;
  padding: 1em;
  border: 1px outset white;
  border-radius: 5px;
  font-size: 0.8em;
}

#registration-form input {
  border-radius: 5px;
}

#registration-header {
  margin: 0;
  margin-left: auto;
  margin-right: auto;
  font-size: 2em;
  padding: 0; 
}

#registration-button {
  margin-left: auto;
  margin-right: auto;
  width: 30%;
}

#login-header {
  margin: 0;
  margin-left: auto;
  margin-right: auto;
  font-size: 2em;
  padding: 0;

}

#registration-form a {
  text-align: center;
}

@media only screen and (min-device-width : 768px){
  #registration-form {
    width: 40%;
    margin-top: 15%;
  }
}

@media only screen and (min-device-width : 1024px){
  #registration-form {
    width: 25%;
    margin-top: 6%;
  }
}