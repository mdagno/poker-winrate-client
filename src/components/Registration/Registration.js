import React from 'react';
import './Registration.css';
import AuthService from '../../services/auth-service';
import { Link } from 'react-router-dom';

export default class Registration extends React.Component {
  state = {
    error: null,
    email: '',
    username: '',
    password: '',
  }

  handleInputChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = e => {
    e.preventDefault()
    const {email, username, password} = this.state;
    AuthService.newUser({email, username, password})
    .then(() => {
      this.props.history.push('/login');
    })
    .catch(res => {
      this.setState({
        error: res.error
      })
    })
  }

  render() {
    return (
      <form id="registration-form" onSubmit={this.handleSubmit}>
        <h3 id="registration-header">Register</h3>
        <label htmlFor="username" className="login-details">E-mail</label>
        <input type="text" className="login-details" name="email" onChange={e => this.handleInputChange(e)}></input>
        <label htmlFor="username" className="login-details">Username</label>
        <input type="text" className="login-details" name="username" onChange={e => this.handleInputChange(e)}></input>
        <label htmlFor="password"className="login-details">Password</label>
        <input type="password" className="login-details" name="password" onChange={e => this.handleInputChange(e)}></input>
        {/* <label htmlFor="confirm-password" className="login-details">Confirm Password</label>
        <input type="password" className="login-details" name="password" onChange={e => this.handleInputChange(e)}></input> */}
        <button type="submit" id="login-button">Register</button>
        {this.state.error && <p>{this.state.error}</p>}
        <Link to="/login">Already have an account?</Link>
      </form>
    )
  }
}