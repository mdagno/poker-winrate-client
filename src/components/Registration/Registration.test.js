import React from 'react';
import Registration from './Registration';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('Registration component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(     
      <BrowserRouter>
          <Registration />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Registration renders correctly', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Registration />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})


