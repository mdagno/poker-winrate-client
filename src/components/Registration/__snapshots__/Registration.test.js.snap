// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component Registration renders correctly 1`] = `
<form
  id="registration-form"
  onSubmit={[Function]}
>
  <h3
    id="registration-header"
  >
    Register
  </h3>
  <label
    className="login-details"
    htmlFor="username"
  >
    E-mail
  </label>
  <input
    className="login-details"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <label
    className="login-details"
    htmlFor="username"
  >
    Username
  </label>
  <input
    className="login-details"
    name="username"
    onChange={[Function]}
    type="text"
  />
  <label
    className="login-details"
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="login-details"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <button
    id="login-button"
    type="submit"
  >
    Register
  </button>
  <a
    href="/login"
    onClick={[Function]}
  >
    Already have an account?
  </a>
</form>
`;
